vpnmanager/vpnservicemanager/vpnprotocol/openvpn/src/main/java/com/kape/openvpn/domain/usecases/HandleOpenVpnHoldOutput.kt

```diff
@@ -22,18 +24,24 @@ import com.kape.openvpn.data.externals.IOpenVpnProcessSocket
 
 internal class HandleOpenVpnHoldOutput(
     private val openVpnProcessSocket: IOpenVpnProcessSocket,
+    private val cache: ICache,
+    private val job: IJob,
 ) : IHandleOpenVpnHoldOutput {
 
     private enum class OpenVpnHoldLinesOfInterest(val line: String) {
-        WAITING_RELEASE("waiting for hold release"),
+        WAITING_RELEASE("waiting for hold release:"),
     }
 
     // region IHandleOpenVpnHoldOutput
     override fun invoke(line: String): Result<Unit> {
         return runCatching {
             when {
                 line.contains(OpenVpnHoldLinesOfInterest.WAITING_RELEASE.line) -> {
-                    openVpnProcessSocket.write("hold release\n").getOrThrow()
+                    val holdReleaseDelaySeconds = line.trim().split(":").last().toLong()
```

Comment:
Should we catch a potential `NumberFormatException` from the toLong()?

Reply:
We can, but we have no way to react if that were to be the case. Performing the `hold release` will lead to a deadlock. If the management interface sends something other than a number, I would say it's an issue on the openvpn side as that's the expected behaviour they are committing to.